be/src/storage/tablet_updates.cpp

```diff
@@ -685,8 +697,90 @@ void TabletUpdates::get_latest_applied_version(EditVersion* latest_applied_versi
     *latest_applied_version = _edit_version_infos[_apply_version_idx]->version;
 }
 
+void TabletUpdates::_apply_empty_rowset_commit(const EditVersionInfo& version_info) {
+    auto manager = StorageEngine::instance()->update_manager();
+    auto tablet_id = _tablet.tablet_id();
+    auto& version = version_info.version;
+    DynamicCache<uint64_t, PrimaryIndex>::Entry* index_entry = NULL;
+    PersistentIndexMetaPB index_meta;
+    bool enable_persistent_index = _tablet.get_enable_persistent_index();
+    if (enable_persistent_index) {
+        index_entry = manager->index_cache().get_or_create(tablet_id);
+        auto& index = index_entry->value();
+        auto st = index.load(&_tablet);
+        manager->index_cache().update_object_size(index_entry, index.memory_usage());
+        if (!st.ok()) {
+            manager->index_cache().remove(index_entry);
+            std::string msg = Substitute("_apply_rowset_commit error: load primary index failed: $0 $1", st.to_string(),
+                                         debug_string());
+            LOG(ERROR) << msg;
+            _set_error(msg);
+            return;
+        }
+        index.prepare(version);
+        st = TabletMetaManager::get_persistent_index_meta(_tablet.data_dir(), tablet_id, &index_meta);
+        if (!st.ok() && !st.is_not_found()) {
+            std::string msg = Substitute("get persistent index meta failed: $0", st.to_string());
+            LOG(ERROR) << msg;
+            _set_error(msg);
+            return;
+        }
+        st = index.commit(&index_meta);
+        if (!st.ok()) {
+            std::string msg = Substitute("primary index commit failed: $0", st.to_string());
+            LOG(ERROR) << msg;
+            _set_error(msg);
+            return;
+        }
+    }
+    {
+        std::lock_guard wl(_lock);
+        // 4. write meta
+        vector<pair<uint32_t, std::shared_ptr<DelVector>>> new_del_vecs;
+        auto st =
+                TabletMetaManager::apply_rowset_commit(_tablet.data_dir(), tablet_id, _next_log_id, version,
+                                                       new_del_vecs, index_meta, _tablet.get_enable_persistent_index());
+        if (!st.ok()) {
+            std::string msg = Substitute("_apply_rowset_commit error: write meta failed: $0 $1", st.to_string(),
```

Comment:
```suggestion
            std::string msg = Substitute("_apply_empty_rowset_commit error: write meta failed: $0 $1", st.to_string(),
```

Reply:
this is on purpose, so grep apply_rowset can show all related logs